x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java

```diff
@@ -22,6 +22,47 @@
 
     static final String NAMESPACE = "elastic";
 
+    private static final ServiceAccount ENTERPRISE_SEARCH_ACCOUNT = new ElasticServiceAccount(
+        "enterprise-search-server",
+        new RoleDescriptor(
+            NAMESPACE + "/enterprise-search-server",
+            new String[] { "manage", "manage_ilm", "manage_index_templates", "manage_security", "monitor" },
```

Comment:
`manage_security` is a very powerful privilege which basically means the account has unbounded privilege. Is it necessary for enterprise search's use case?

Reply:
Enterprise Search needs to be able to create/modify native users and Elasticsearch roles.
Is there a different privilege that could be used for this?

Reply:
Thanks for clarification. In this case, `manage_security` is the privilege to go. We don't have separate privileges for working with users and roles only. 

I understand this must be how Enterprise Search always works. But it would be great if we could learn about its details and potentially work out something that allows us moving away from `manage_security` in future.

Reply:
Another point is about the `manage` privilege, it covers all `manage_ilm`, `manage_index_templates` and `monitor` (but _not_ `manage_security`). It is also a powerful privilege. What actions do Enterprise Search need it for?

Reply:
`manage_ilm` - Enterprise Search creates several data streams for logging events, for example analytics events or API logs. These data streams have an ILM policy that is created by Enterprise Search.

`manage_index_templates` - we use composable index templates for these data streams to ensure that they are ECS compatible

`monitor` - Enterprise Search needs cluster information, such as the cluster health and node information. For example, when Enterprise Search starts, it waits for the cluster to be healthy. Node information is useful to determine what plugins are available, for example we use `analysis-icu` if it is available.

> I understand this must be how Enterprise Search always works. But it would be great if we could learn about its details and potentially work out something that allows us moving away from manage_security in future.

I agree. Is there an issue if we remove this cluster privilege for the service account some time in future when Enterprise Search no longer needs it? I don't think it is, but just asking to be sure.

Reply:
> I agree. Is there an issue if we remove this cluster privilege for the service account some time in future when Enterprise Search no longer needs it? I don't think it is, but just asking to be sure.

If we don't have Enterprise Search on older versions talking to ES on new versions with reduced privieges, it should not be an issue.

Reply:
@ioanatia Thanks for clarifing the intended usage for `manage_ilm`, `manage_index_templates` and `monitor`. Do they already cover all that is necessary for Enterprise Search? Sorry if it wasn't clear, but my original question was whether we can drop `manage` if there is nothing else required outside of the three more fine-grained privileges. Examples of extra privileges granted by `manage` are `manage_ml`, `manage_watcher`, `manage_slm`, `manage_pipeline` etc. Do Enterprise Searh need any of these extra privileges?